### PR TITLE
Allow pagination for Ruby SDK

### PR DESCRIPTION
**Pull request by choran**
This is related to https://github.com/intercom/intercom-ruby/issues/215
Previously we were returning all of the elements in a request even if the user specified a single page of results. Just adding in a check here to monitor the current page and return that page if that is what the user specified. This will allow users to specific the per page request number and the specific page they want returned.

### CODE REVIEW

File: spec/unit/intercom/client_collection_proxy_spec.rb [hunk 1]
```diff
@@ -32,4 +32,49 @@
...
+  end
+
+  it "works with asc order" do
+    test_date=1457337600
+    time_increment=1000
+    users = [test_user_dates(email="user1@example.com", created_at=test_date), test_user_dates(email="user2@example.com", created_at=test_date-time_increment),
+             test_user_dates(email="user3@example.com", created_at=test_date-2*time_increment), test_user_dates(email="user4@example.com", created_at=test_date-3*time_increment)]
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at"}).returns(users_pagination(true, per_page=2, page=30, total_pages=2, total_count=3, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at").map {|user| user.email }
+    result.must_equal %W(user1@example.com user2@example.com user3@example.com user4@example.com)
```
Comment (Skaelv) [inline spec/unit/intercom/client_collection_proxy_spec.rb]: user per page = 2 but you have 4 emails in the results ?

File: spec/unit/intercom/client_collection_proxy_spec.rb [hunk 2]
```diff
@@ -32,4 +32,49 @@
...
+    result.must_equal %W()
+  end
+
+  it "works with asc order" do
+    test_date=1457337600
+    time_increment=1000
+    users = [test_user_dates(email="user1@example.com", created_at=test_date), test_user_dates(email="user2@example.com", created_at=test_date-time_increment),
+             test_user_dates(email="user3@example.com", created_at=test_date-2*time_increment), test_user_dates(email="user4@example.com", created_at=test_date-3*time_increment)]
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at"}).returns(users_pagination(true, per_page=2, page=30, total_pages=2, total_count=3, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at").map {|user| user.email }
```
Comment (Skaelv) [inline spec/unit/intercom/client_collection_proxy_spec.rb]: `result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at").map(&:email)`
should work as well

File: spec/unit/intercom/client_collection_proxy_spec.rb (hunk 1 above)
Comment (choran) [inline spec/unit/intercom/client_collection_proxy_spec.rb]: cool, will update it. for to change that for asc and desc tests

File: spec/unit/intercom/client_collection_proxy_spec.rb (hunk 2 above)
Comment (choran) [inline spec/unit/intercom/client_collection_proxy_spec.rb]: ok, I can change one of the unit tests to use the (&:email) format

Comment (Skaelv) [inline spec/unit/intercom/client_collection_proxy_spec.rb]: I think it's just faster as you won't load the users in the memory to get their emails but will just load the emails

**Comment by Skaelv** [comment]
squash those two commits in one and you are ready to merge!
🚢

**Pull request merged**